flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/FailedCheckpointStats.java

```diff
@@ -87,15 +72,20 @@
             @Nullable SubtaskStateStats latestAcknowledgedSubtask,
             @Nullable Throwable cause) {
 
-        super(checkpointId, triggerTimestamp, props, totalSubtaskCount, taskStats);
+        super(
+                checkpointId,
+                triggerTimestamp,
+                props,
+                totalSubtaskCount,
+                numAcknowledgedSubtasks,
+                taskStats,
+                PendingCheckpointStatsCallback.noOp(),
```

Comment:
I don't understand this. How is this supposed to be working? How stats are being reported for failed checkpoints? How are we reporting I get, that in this commit you would like to have temporary `noOp` being used, but I don't see this being replaced later? Is something missing, or is this a dead code, or am I missing something?

Reply:
First, `PendingCheckpointStats` is created. It has "normal" callbacks and subtasks reports update it as usual.
Upon failure it is converted to a `FailedCheckpointStats` using one of these "normal" callbacks.
This new object won't be converted so it has `noOp` callbacks. It still can be updated from subtasks.

I tried to refactor this code to have just one `CheckpointStats` class without conversions and callbacks.
But the change was to big compared to the functionality added so I dropped it.

Reply:
I replaced `noOp` with a static one throwing `UnsupportedOperationException`.